omnigibson/object_states/temperature.py

```diff
@@ -41,17 +44,30 @@ def _set_value(self, new_value):
         return True
 
     def _update(self):
+        # Avoid circular import
+        from omnigibson.object_states.on_fire import OnFire
+
         # Start at the current temperature.
         new_temperature = self.value
 
         # Find all heat source objects.
         affected_by_heat_source = False
-        for obj2 in self._simulator.scene.get_objects_with_state(HeatSourceOrSink):
-            # Obtain heat source position.
-            heat_source = obj2.states[HeatSourceOrSink]
-            heat_source_state, heat_source_position = heat_source.get_value()
+        heat_sources = og.sim.scene.get_objects_with_state(HeatSourceOrSink) | \
```

Comment:
maybe make this an external utility call? That way we can easily find it and add to it if we add other heat sources in the future

Reply:
I added the get_objects_with_state_recursive function call. I think it's safe to assume (at least for now) that all heat source related states would be subclasses of `HeatSourceOrSink`.